### DISCUSSION TITLE
How can I make the header guard case insensitive?

### DISCUSSION THREAD

**Discussion by ghost** [actix/actix-web#2445] [Q&A]
I have a route with a header guard like this.  It works but it is case sensitive - headers should not be case sensitive.  Can anyone suggest to me please how I make this case insensitive?  thanks!

```
                .route(
                    web::post()
                        .guard(guard::Header("content-type", "application/json"))
                        .to(upload_to_hashmap_json),
                )
```

**Comment by CrayonChimp** [comment]
It's weird because usually http header is not case sensitive.
![image](https://user-images.githubusercontent.com/9478533/142396030-f907bb88-81e1-4579-9a2b-e21cf4a6a745.png)
The test also confirms my assumption.
Anyway, maybe `guard::fn_guard` is what you need
```Rust
.guard(guard::fn_guard(|head| {
    //do some transfer
    bool
}))
```

**Comment by robjtede** [comment]
> headers should not be case sensitive

Specifically header names are case insensitive. If you're seeing the name being handled in a case sensitive way, please open a bug report.

If you want case insensitive header value guards, that's a feature request or use the fn_guard method suggested by @bellemere.
